ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -28,6 +28,8 @@ def initialize(
     # @param expected_version [:any, :auto, :none, Integer] controls optimistic locking strategy. {http://railseventstore.org/docs/expected_version/ Read more}
     # @return [self]
     def publish(events, stream_name: GLOBAL_STREAM, expected_version: :any)
+      assert_nil_events(events)
```

Comment:
@pawelpacana, @jandudulski Why not change it around, instead to explicitly check some negative examples, why not ask `events` *is* an `Array`?

Reply:
Because it can be just an event or array of events

Reply:
Than check for is array or is an event. But not explicitly do a partial negative check. There are so many other invalid values you do not check for.

`false`, `true`, `Object.new`, the list of "incompatible" inputs is endless, the list of compatible ones finite.

Reply:
I would prefer allow list too but it is not that straightforward. I’m fine accepting Object.new if it responds to right methods.

Reply:
Ah, now I got your point @mbj. I like this point of view, the issue is that `event` might be many things, but I'll figure it out.